### DISCUSSION TITLE
Transparent windows?

### DISCUSSION THREAD

**Discussion by srijan-paul** [slint-ui/slint#5111] [Q&A]
I'm making a screen capture app, and would like to have transparent windows so the user can draw over a screenshot.
Is there any way to achieve this in Slint?

Here is a prototype of the app in Tauri that I want to port to Slint:
https://github.com/slint-ui/slint/assets/50487716/b54a850f-eaca-4d19-a948-c71d1256f90d

**Comment by ogoffart** [comment]
You can have a `Window` with `background: transparent;` and that should work. You can also use it with `no-frame: true;`

**Reply by LexiconCode** [reply]
With transparency is possible to also have click-through with mouse and keyboard events without changing foreground windows?

**Reply by ogoffart** [reply]
That's something more than just window transparency. You will have to use private API to achieve that. (eg, using the i-slint-backend-winit crate and winit::window::set_cursor_hittest)

**Reply by filen1** [reply]
I also ran into a need for click-through. Works fine with backend crates.

**Reply by LexiconCode** [reply]
> I also ran into a need for click-through. Works fine with backend crates.

It would be nice to have cross-platform abstraction for click through.

**Reply by m4heshd** [reply]
@LexiconCode Is click-through still not possible?

**Reply by LexiconCode** [reply]
> @LexiconCode Is click-through still not possible?

Not without digging into an OS-specific package or a third-party package looking at today's documentation. They still haven't implemented click through transparency as a OS agnostic abstraction.

**Reply by m4heshd** [reply]
> Not without digging into an OS-specific package or a third-party package looking at today's documentation. They still haven't implemented click through transparency as a OS agnostic abstraction.

Dang, that's unfortunate. I'm just shopping around for a Tauri alternative now that the v2 has destroyed all things good about Tauri, and this feature is a necessity for me.

**Reply by LuminairPrime** [reply]
Is there a feature request for click-through that I'm missing, or should we create one?

**Comment by srijan-paul** [comment]
Thank you, @ogoffart !

BTW, I scoured the docs but couldn't find anything that would help me build an image editor (lines, circles, curves, polygons, etc).
Is there a canvas like API I can use?

It seems that the `Image` object can be used for this, but only if I directly write raw bytes to an image buffer.

**Reply by ogoffart** [reply]
You can use Image, you can also use the Path element for arbitrary shape.

You can also have a look to the ["circledraw" example](https://github.com/slint-ui/slint/tree/master/examples/7guis#circle-drawer)

**Comment by Horbin-Magician** [comment]
I'm similarly working on screenshot related features, so you can refer to my project if you'd like: https://github.com/Horbin-Magician/rotor.
If your code is open source, I'd love to learn your relevant code too!
